### ISSUE TITLE
Error Response don't work 

### ISSUE THREAD

**Issue by DjCodeGit** [sailscastshq/boring-stack#55]
Hello,

When i try to do this in action 2: 
```js
sails.inertia.share('errors', {
        password: errorMessage
      })
      throw { badCombo: 'back' }
```

In the front "Vuejs", the inertia's form trigger onSuccess evenement 😢 
```js
form.post('/login',{
    onSuccess:()=>{
      console.log('success:',form.errors)
    },
    onError:()=>{
      console.log('error:',form.errors)
    }
  })
```

And form.errors has no keys....

**Comment by DominusKelvin** [comment]
Thanks for this, @DjCodeGit in the latest version of the Mellow template I think I fixed this for errors. You can look at the code in the signup and login actions

**Comment by DjCodeGit** [comment]
Hello @DominusKelvin , thanks for your answer. Unfortunately I had already tested the Mellow template and I got this error in case of user not found. 
![image](https://github.com/sailscastshq/boring-stack/assets/37690211/d3952aa0-c024-48e3-bac2-9a6d765cd886)

**Comment by DjCodeGit** [comment]
Why add new issue? This is the same problem for me. I would like to manage the backend's Error and send it to the front-end using "onError()" method of the Inertia's form.

**Comment by DominusKelvin** [comment]
Oh got it I thought it was a new error gotten from a user not being found.

I'm working on error handling for a patch release of mellow-vue which will also close this issue.

I'll let you know as soon as I make that fix

**Comment by DominusKelvin** [comment]
Hey @DjCodeGit I have been able to reproduce and I'll cut a new patch with these changes soon.

![CleanShot 2023-09-29 at 14 41 55@2x](https://github.com/sailscastshq/boring-stack/assets/24433274/ba6cff82-1802-407e-ae3c-b8f9615bd9e1)

**Comment by DominusKelvin** [comment]
Hey @DjCodeGit were you able to move forward with this? I have been caught up with other things and I haven't add time to put in the error checks.

**Comment by davidtwin** [comment]
Hi Kelvin, did you manage to fix the error handling mellow-vue?

**Comment by DominusKelvin** [comment]
Hey @davidtwin, error handling exists today, I just haven't added an example in the template. What sort of errors do you want to handle? Maybe I can point you in the right direction.

**Comment by davidtwin** [comment]
Hey @DominusKelvin 
When i write this on my controller

sails.inertia.share('errors', {
        email: credentialErrorMessage
      })

nothing i getting saved onto errors.

on your pingcrm example it saves on to it but not on this template

**Comment by DominusKelvin** [comment]
Oh I see. I’ll share a little code snippet that should unblock you on
error handling. This also look like a good candidate for the docs when I
write it.

Kelvin Oghenerhoro Omereshone
CEO, The Sailscasts Company
Ship less JavaScript™️ <https://shiplessjavascript.com>
consultant
Building Sailscasts <https://sailscasts.com/> & Hagfish
<https://sailscasts.com/hagfish>

Say hi on
Twitter(X) <https://twitter.com/Dominus_Kelvin>

On January 24, 2024, GitHub ***@***.***> wrote:
> Hey @DominusKelvin <https://github.com/DominusKelvin>
>  When i write this on my controller
>
> sails.inertia.share('errors', {
>  email: credentialErrorMessage
>  })
>
> nothing i getting saved onto errors.
>
> on your pingcrm example it saves on to it but not on this template
>

—
Reply to this email directly, view it on GitHub
<https://github.com/sailscastshq/boring-stack/issues/55#issuecomment-
1907242636>, or unsubscribe
<https://github.com/notifications/unsubscribe-
auth/AF2NE6UD6WRVCKVAFCDLATDYQBWGNAVCNFSM6AAAAAA5ADLFQOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMBXGI2DENRTGY>.
You are receiving this because you were mentioned.Message ID:
***@***.***>

**Comment by DominusKelvin** [comment]
Hey @davidtwin and @DjCodeGit this #80 resolves this and I will update the docs on how to send server side validation errors to your SPA via props.

**Comment by DominusKelvin** [comment]
The new release fixes this issue @DjCodeGit :)

https://github.com/sailscastshq/boring-stack/discussions/83

**Comment by DjCodeGit** [comment]
Thanks you Kelvin. I'll test it as soon as possible.Le 22 févr. 2024 09:54, Kelvin Oghenerhoro Omereshone ***@***.***> a écrit :
The new release fixes this issue @DjCodeGit :)
#83

**Comment by DominusKelvin** [comment]
> Thanks you Kelvin. I'll test it as soon as possible.Le 22 févr. 2024 09:54, Kelvin Oghenerhoro Omereshone ***@***.***> a écrit :
> 
> The new release fixes this issue @DjCodeGit :)
> 
> #83
> 
> 
> 
> —Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you were mentioned.Message ID: ***@***.***>

Awesome thanks for your patience on this one. I needed a missing piece to really have the best DX for server side validation errors - Sails Flash

When you upgrade just instal sails-flash in your project as well upgrade inertia-sails

I'll write you  release article on The Sailscasts Blog soon
